api/v1beta1/openstackdataplanedeployment_types.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"fmt"
+
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OpenStackDataPlaneDeploymentSpec defines the desired state of OpenStackDataPlaneDeployment
+type OpenStackDataPlaneDeploymentSpec struct {
+
+	// +kubebuilder:validation:Required
+	// NodeSets is the list of NodeSets deployed
+	NodeSets []string `json:"nodeSets"`
+
+	// AnsibleTags for ansible execution
+	// +kubebuilder:validation:Optional
+	AnsibleTags string `json:"ansibleTags,omitempty"`
+
+	// AnsibleLimit for ansible execution
+	// +kubebuilder:validation:Optional
+	AnsibleLimit string `json:"ansibleLimit,omitempty"`
+
+	// AnsibleSkipTags for ansible execution
+	// +kubebuilder:validation:Optional
+	AnsibleSkipTags string `json:"ansibleSkipTags,omitempty"`
```

Comment:
I could make that change. I didn't initially go that route since these fields feel more applicable to a deployment, it didn't feel necessary to nest or scope them under an ansible field. Let's see if there is additional feedback from others.